modules/sdk-coin-avaxp/src/lib/transactionBuilderFactory.ts

```diff
@@ -17,19 +19,46 @@ export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
   /** @inheritdoc */
   from(raw: string): TransactionBuilder {
     utils.validateRawTransaction(raw);
-    const tx = new Tx();
-    tx.fromBuffer(BufferAvax.from(raw, 'hex'));
+    let tx: Tx | EVMTx;
     let transactionBuilder: TransactionBuilder;
-    if (ValidatorTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
-      transactionBuilder = this.getValidatorBuilder();
-    } else if (ExportTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
-      transactionBuilder = this.getExportBuilder();
-    } else if (ImportTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
-      transactionBuilder = this.getImportBuilder();
-    } else {
-      throw new NotSupported('Transaction cannot be parsed or has an unsupported transaction type');
-    }
+    try {
+      tx = new Tx();
+      // could throw an error if a txType doesn't match.
+      tx.fromBuffer(BufferAvax.from(raw, 'hex'));
+
+      const sameBlockchainId =
+        utils.cb58Encode(tx.getUnsignedTx().getTransaction().getBlockchainID()) ===
+        (this._coinConfig.network as AvalancheNetwork).blockchainID;
+      if (!sameBlockchainId) {
+        throw new Error('It is not a transaction of this network');
+      }
+
+      if (ValidatorTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
+        transactionBuilder = this.getValidatorBuilder();
+      } else if (ExportTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
+        transactionBuilder = this.getExportBuilder();
+      } else if (ImportTxBuilder.verifyTxType(tx.getUnsignedTx().getTransaction())) {
+        transactionBuilder = this.getImportBuilder();
+      } else {
+        throw new NotSupported('Transaction cannot be parsed or has an unsupported transaction type');
+      }
+    } catch {
+      tx = new EVMTx();
+      tx.fromBuffer(BufferAvax.from(raw, 'hex'));
 
+      const sameBlockchainId =
+        utils.cb58Encode(tx.getUnsignedTx().getTransaction().getBlockchainID()) ===
+        (this._coinConfig.network as AvalancheNetwork).cChainBlockchainID;
+      if (!sameBlockchainId) {
+        throw new Error('It is not a transaction of this network or C chain');
+      }
```

Comment:
```suggestion
validateBlockChainId(tx)
```

Reply:
it'll be done in next pr
https://github.com/BitGo/BitGoJS/blob/b3805da4d5e69b358be6ad5e813e5e57b83b3c0e/modules/sdk-coin-avaxp/src/lib/transactionBuilderFactory.ts#L30